server/data/data-file/src/main/java/org/apache/james/user/file/UsersFileRepository.java

```diff
@@ -148,18 +153,18 @@ public synchronized User getUserByName(String name) throws UsersRepositoryExcept
      * @return The real name
      * @throws UsersRepositoryException
      */
-    private String getRealName(String name, boolean ignoreCase) {
+    private Username getRealName(Username name, boolean ignoreCase) {
         if (ignoreCase) {
-            Iterator<String> it = list();
+            Iterator<Username> it = list();
             while (it.hasNext()) {
-                String temp = it.next();
-                if (name.equalsIgnoreCase(temp)) {
+                Username temp = it.next();
+                if (name.asString().equalsIgnoreCase(temp.asString())) {
```

Comment:
will need to be fixed if we want to handle Locale

Reply:
added a FIXME comment